### ISSUE TITLE
Microsoft ATP - URL Encoding Breaks Email Reporting Function

### ISSUE THREAD

**Issue by CMS009** [gophish/gophish#1975]
What version of Gophish are you using?: 0.11.0

Brief description of the issue: When using Microsoft ATP, it's URL encoding seems to break the Email reporting feature.  Or if it's looking at X-Headers, I've not determined which one is the culprit.

What are you expecting to see happen? :  When a user reports a simulated phishing email that's been processed (URL encoded) by Microsoft ATP, it should show (on the GoPhish campaign page) as successfully reported by the user.

What are you seeing happen?  The email is processed (read by GoPhish using IMAP), but does not show as reported in the campaign.

Example of MS ATP encoded format:

```
https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgophish.example.com%2F%3Frid%3DYbkD7ke&amp;data=02%7C01%7Cuser%40example.com%7Cc44588be037dfcje93gsh29s85b20b877%7C77aa56fb226e4666a99ea1c1b3fc9551%7C0%7C0%7C637359526304827107&amp;sdata=qd8uZgv43w0dkd0274gbh1049fn5lH%2FLhBkLfUtiKeA%3D&amp;reserved=0
```

Example of MS ATP UTF-8 decoded format:

```
https://nam10.safelinks.protection.outlook.com/?url=https://gophish.example.com/?rid=YbkD7ke&amp;data=02|01|user@example.com|c44588be037dfcje93gsh29s85b20b877|77aa56fb226e4666a99ea1c1b3fc9551|0|0|637359526304827107&amp;sdata=qd8uZgv43w0dkd0274gbh1049fn5lH/LhBkLfUtiKeA=&amp;reserved=0
```

Please provide as many steps as you can to reproduce the problem:
It seems if I turn off the Microsoft ATP feature, the plain unencoded URL works (reports as expected).

**Comment by glennzw** [comment]
Hi @CMS009, thanks for getting in touch. Would you be able to share the raw .eml file to help me figure this one out? You could upload it here as an attachment, or forward it to me as an attachment (glennzw AT protonmail DOT com).

**Comment by jordan-wright** [comment]
Fixed via #1976. If you continue to run into issues, just let us know and we can revisit that fix :)

**Comment by CMS009** [comment]
@glennzw It works like a champ!  Thanks.
